dashboard/src/components/ManageVersionModal.tsx

```diff
@@ -0,0 +1,106 @@
+import { Delete } from '@mui/icons-material';
+import Add from '@mui/icons-material/Add';
+import Button from '@mui/joy/Button';
+import DialogContent from '@mui/joy/DialogContent';
+import DialogTitle from '@mui/joy/DialogTitle';
+import FormControl from '@mui/joy/FormControl';
+import Input from '@mui/joy/Input';
+import Modal from '@mui/joy/Modal';
+import ModalDialog from '@mui/joy/ModalDialog';
+import Stack from '@mui/joy/Stack';
+import { ChangeEvent, FC, FormEvent, useState } from 'react';
+import { useDeleteVersion } from '../api/hooks/useDeleteVersion';
+import { usePostVersion } from '../api/hooks/usePostVersion';
+
+type ManageVersionModalProps = {
+  projectId: number;
+  versionId?: number;
+  versionName?: string;
+}
+
+export const ManageVersionModal: FC<ManageVersionModalProps> = ({ projectId, versionId, versionName }) => {
+  const [open, setOpen] = useState(false);
+  const [newProjectName, setNewProjectName] = useState('');
+
+  const { mutate: deleteVersion } = useDeleteVersion(projectId, versionId!);
+  const { mutate: postNewVersion } = usePostVersion(projectId);
+
+  const onProjectNameChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const regex = new RegExp('^[a-zA-Z0-9_\\-\\.]+$');
+
+    if (!regex.test(event.target.value)) {
+      return;
+    }
+
+    setNewProjectName(event.target.value);
+  }
+
+  const onSubmit = (event: FormEvent<HTMLFormElement>) => {
+    if (versionId) {
+      deleteVersion();
+    } else {
+      postNewVersion({ name: newProjectName });
+    }
+
+    event.preventDefault();
+    setOpen(false);
```

Comment:
```suggestion
    event.preventDefault();
    
    if (versionId) {
      deleteVersion();
    } else {
      postNewVersion({ name: newProjectName });
    }

    setOpen(false);
```

Good idea to start with preventing the event propagating